ibm_cloud_sdk_core/base_service.py

```diff
@@ -277,6 +277,11 @@ def send(self, request: requests.Request, **kwargs) -> DetailedResponse:
         # Check to see if the caller specified the 'stream' argument.
         stream_response = kwargs.get('stream') or False
 
+        # Remove the keys we set manually, don't let the user to overwrite these.
+        reserved_keys = ['method', 'url', 'headers', 'params', 'cookies']
+        for key in reserved_keys:
+            kwargs.pop(key, None)
```

Comment:
I wonder if we want to just throw an error here or a message so the user knows their argument is being omitted.

Reply:
I think throwing an exception is not a good idea, since removing the headers will be quite common. I've added a warning log for now.